roles/sushy_emulator/tasks/setup_access.yml

```diff
@@ -0,0 +1,55 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Look for Sushy Emulator keypair file
+  register: _sushy_emulator_key_file
+  ansible.builtin.stat:
+    path: "{{ (cifmw_sushy_emulator_resource_directory, 'sushy_emulator_key') | path_join }}"
+
+- name: "Generate SSH keypair"
+  when:
+    - not _sushy_emulator_key_file.stat.exists
+  community.crypto.openssh_keypair:
+    comment: "sushy emulator SSH keypair"
+    path: "{{ (cifmw_sushy_emulator_resource_directory, 'sushy_emulator_key') | path_join }}"
+    type: "ecdsa"
+
+- name: Set authorized key taken from file
+  delegate_to: "{{ cifmw_edpm_deploy_baremetal_hypervisor_target }}"
+  become: true
+  ansible.posix.authorized_key:
+    user: "{{ cifmw_sushy_emulator_libvirt_user }}"
+    state: present
+    key: "{{ lookup('file', cifmw_sushy_emulator_resource_directory+'/sushy_emulator_key.pub') }}"
+
+- name: Base64 encode ssh private key
+  ansible.builtin.slurp:
+    src: "{{ cifmw_sushy_emulator_resource_directory }}/sushy_emulator_key"
+  register: _cifmw_sushy_emulator_private_key_b64
+  no_log: true
+
+- name: Base64 encode ssh public key
+  ansible.builtin.slurp:
+    src: "{{ cifmw_sushy_emulator_resource_directory }}/sushy_emulator_key.pub"
+  register: _cifmw_sushy_emulator_public_key_b64
+  no_log: true
+
+- name: Run ssh-keyscan
+  ansible.builtin.shell:
+    cmd: |
+      set -o pipefail
+      ssh-keyscan -H {{ hostvars[cifmw_edpm_deploy_baremetal_hypervisor_target].ansible_host }} 2>/dev/null | base64
```

Comment:
@viroel had issues with FIPS enabled system and ssh-keyscan.

Reply:
Do we expect to use FIPs image for the hypervisor that will host the baremetal vms?

If so, could you or Doug point me to the workaround :)

Reply:
I'm not sure, to be fair - and I don't remember where he switched the way to get it right, you probably want to check with him directly :)

Reply:
The issue with FIPS was a invalid cypher configuration (for FIPS) in ssh config - so the workaround wasn't merged. You should be ok to use ssh-keyscan.